### DISCUSSION TITLE
construction graph visualizer

### DISCUSSION THREAD

**Discussion by deltanedas** [space-wizards/space-station-14#21629] [Technical]
i want to do basic pizza construction but i have no idea how to visualize the pizza being made
general idea is have containers for sauce toppings etc on flat dough and somehow visualize them for sauce layer / cheese layer / meat layer etc

@mirrorcult maybe you know

**Comment by metalgearsloth** [comment]
She just started her break today.

Containers might be easiest but most expensive in terms of entity cost.

**Reply by deltanedas** [reply]
:trollface:

i plan to make it so flat dough on its own has no containers and a construction step makes it a pizza base that has containers, would that avoid costs

**Reply by Emisse** [reply]
we should probably make something generic so this can be applied to other customizable foods like salads, soups, sandwiches, etc

**Reply by deltanedas** [reply]
visualizer or the container thing

i dont think construction graphs can be generic over different item types (e.g. pizza base + PizzaTopping tag items vs bowl with boiling water and SoupIngredient tag items)

**Reply by Emisse** [reply]
visualizer

**Reply by deltanedas** [reply]
yeah that should be generic

**Comment by deltanedas** [comment]
@metalgearsloth related to pizza how should i set a pizzas sprite to a recipe pizza's sprite
A) raise a network event in pvs range for clients to copy prototype's sprite to the pizza entity's sprite
B) add a networked component that makes clients do above
C) spawn the recipe pizza and set its solution to be whatever the pizza contains then delete the old entity

**Comment by PJB3005** [comment]
Robust editor.....

**Reply by deltanedas** [reply]
what
